File: local/bin/post_twitter.js
var fs = require('fs'),
    twitter = require('ntwitter'),
    ini = require('ini'),
    Hebcal = require('hebcal');

var iniPath = '/home/hebcal/local/bin/hebcal-dot-com.ini',
    config = ini.parse(fs.readFileSync(iniPath, 'utf-8'));

function upcomingDow(searchingDow) {
    var today = new Date(),
        todayDow = today.getDay();
    if (searchingDow == todayDow) {
        return today;
    }
    var upcoming = new Date(today);
    upcoming.setHours(0, 0, 0, 0);
    upcoming.setDate(upcoming.getDate() + (searchingDow - todayDow));
    if (searchingDow < todayDow) {
        upcoming.setDate(upcoming.getDate() + 7);
    }
    return upcoming;
}

var saturday = upcomingDow(6);
console.log(saturday);
var parashah = new Hebcal.HDate(saturday).getSedra().join('-');

var twitterStatus = 'This week\'s #Torah portion is Parashat ' + parashah;
console.log(twitterStatus);

var twit = new twitter({
  consumer_key: config['hebcal.twitter.consumer_key'],
  consumer_secret: config['hebcal.twitter.consumer_secret'],
  access_token_key: config['hebcal.twitter.token'],
  access_token_secret: config['hebcal.twitter.token_secret']
});

twit.updateStatus(twitterStatus,
    function (err, data) {
        console.log(data);
        process.exit();
    }
);

File: local/bin/post_twitter.py
#!/usr/bin/python3

import sys
import tweepy
import configparser
import io

config = configparser.RawConfigParser()

f = open('/home/hebcal/local/etc/hebcal-dot-com.ini', 'r')

output = io.StringIO()
output.write('[DEFAULT]\n')
output.write(f.read())
output.seek(0,0)

contents = output.getvalue()

config.read_string(contents)

CONSUMER_KEY = config.get('DEFAULT', 'hebcal.twitter.consumer_key')
CONSUMER_SECRET = config.get('DEFAULT', 'hebcal.twitter.consumer_secret')
ACCESS_KEY = config.get('DEFAULT', 'hebcal.twitter.access_key')
ACCESS_SECRET = config.get('DEFAULT', 'hebcal.twitter.access_secret')

auth = tweepy.OAuthHandler(CONSUMER_KEY, CONSUMER_SECRET)
auth.set_access_token(ACCESS_KEY, ACCESS_SECRET)
api = tweepy.API(auth)
api.update_status(sys.argv[1])
